### PR TITLE
refactor: move build over to mill

### PR DESCRIPTION
**Pull request by ckipp01**
Wanted to see if I could use this repo as a testing ground for a Mill plugin I just wrote, which is a Mill replacement of the sbt g8 plugin. It does the same exact thing, but since this is a Mill template I figured, why not! This also adds in some targets to not only validate that the project can generate, but also that those targets can work after generation.

I also add in a license which g8 mentions is important http://www.foundweekends.org/giter8/template.html#Use+CC0+1.0+for+template+licensing.

### CODE REVIEW

File: src/main/g8/$moduleName$/test/src/example/HelloSuite.scala [hunk 1]
```diff
@@ -3,11 +3,11 @@ package example
 class HelloSuite extends munit.FunSuite {
   test("numbers") {
     val obtained = 42
-    val expected = 43
```
Comment (tgodzik) [inline src/main/g8/$moduleName$/test/src/example/HelloSuite.scala:6]: This was kind of supposed to show a broken test :sweat_smile:

**Review by tgodzik — APPROVED** [review]
I still have issues understanding how to work with mill, but this makes perfect sense here. Thanks!

File: src/main/g8/$moduleName$/test/src/example/HelloSuite.scala (hunk 1 above)
Comment (ckipp01) [reply src/main/g8/$moduleName$/test/src/example/HelloSuite.scala:6]: Ah! I can take out the check for `__.test` if you'd rather keep it this way? ha, I just wanted to show that the validation targets work. I was also thinking of adding a way to allow for a target to fail when expected, but I figured most templates wanted everything working out of the box.

Comment (tgodzik) [reply src/main/g8/$moduleName$/test/src/example/HelloSuite.scala:6]: It's totally fine, it's really not important, people can easily figure out.

**Pull request merged**